src/Avalonia.Controls/Shapes/Shape.cs

```diff
@@ -61,12 +61,26 @@ public Geometry RenderedGeometry
         {
             get
             {
-                if (_renderedGeometry == null)
+                if (_renderedGeometry == null && DefiningGeometry != null)
                 {
-                    if (DefiningGeometry != null)
+                    if (_transform == Matrix.Identity)
+                    {
+                        _renderedGeometry = DefiningGeometry;
+                    }
+                    else
                     {
                         _renderedGeometry = DefiningGeometry.Clone();
-                        _renderedGeometry.Transform = new MatrixTransform(_transform);
+
+                        if (_renderedGeometry.Transform == null ||
+                            _renderedGeometry.Transform.Value == Matrix.Identity)
+                        {
+                            _renderedGeometry.Transform = new MatrixTransform(_transform);
+                        }
+                        else
+                        {
+                            _renderedGeometry.Transform = new MatrixTransform(
+                                _renderedGeometry.Transform.Value * _transform);
```

Comment:
`_renderedGeometry.Transform` seems to be changed on each call to the getter if `_transform` is not null or `Matrix.Identity`. Is that intended?

Reply:
Sorry, I don't follow. `_renderedGeometry.Transform` will only be set when `_renderedGeometry` has been cleared to null, in which case the `DefiningGeometry` will be cloned and it's `Transform` set. Or am I missing something obvious?

Reply:
Ah, I've missed that bit, sorry.